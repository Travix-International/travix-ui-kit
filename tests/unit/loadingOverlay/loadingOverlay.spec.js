import { shallow } from 'enzyme';
import React from 'react';
import LoadingOverlay from '../../../components/loadingOverlay/loadingOverlay';

describe('LoadingOverlay', () => {
  describe('#render()', () => {
    it('render basic loading overlay', () => {
      const wrapper = shallow(
        <LoadingOverlay>
          Some data
        </LoadingOverlay>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('render loading overlay with custom class', () => {
      const wrapper = shallow(
        <LoadingOverlay className="my-custom-class">
          Some data
        </LoadingOverlay>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('render loading overlay with transparency', () => {
      const wrapper = shallow(
        <LoadingOverlay className="my-custom-class" transparency>
          Some data
        </LoadingOverlay>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('render loading overlay in loading mode', () => {
      const wrapper = shallow(
        <LoadingOverlay className="my-custom-class" loading>
          Data is loading...
        </LoadingOverlay>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('render loading overlay with transparency in loading mode', () => {
      const wrapper = shallow(
        <LoadingOverlay className="my-custom-class" loading transparency>
          Data is loading... You can see that message.
        </LoadingOverlay>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('render loading overlay with left message direction', () => {
      const wrapper = shallow(
        <LoadingOverlay className="my-custom-class" loading messageDirection="left">
          Data is loading...
        </LoadingOverlay>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('render loading overlay with loading message', () => {
      const wrapper = shallow(
        <LoadingOverlay className="my-custom-class" loading message="Loading your data...">
          Data is loading...
        </LoadingOverlay>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('render without spinner', () => {
      const wrapper = shallow(
        <LoadingOverlay className="my-custom-class" loading spinner={false}>
          Data is loading...
        </LoadingOverlay>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('render without children - should return null', () => {
      const wrapper = shallow(
        <LoadingOverlay className="my-custom-class" loading message="Loading your data..." />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
