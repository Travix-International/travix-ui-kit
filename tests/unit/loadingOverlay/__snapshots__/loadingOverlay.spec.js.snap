// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlay #render() render basic loading overlay 1`] = `
<div
  className="ui-loading-overlay"
>
  <div
    className="ui-loading-overlay__content"
  >
    Some data
  </div>
</div>
`;

exports[`LoadingOverlay #render() render loading overlay in loading mode 1`] = `
<div
  className="ui-loading-overlay ui-loading-overlay_loading my-custom-class"
>
  <div
    className="ui-loading-overlay__loading-container"
  >
    <[object Object] />
    <span
      classNames="ui-loading-overlay__loading-message ui-loading-overlay__loading-message_right"
    />
  </div>
  <div
    className="ui-loading-overlay__content"
  >
    Data is loading...
  </div>
</div>
`;

exports[`LoadingOverlay #render() render loading overlay with custom class 1`] = `
<div
  className="ui-loading-overlay my-custom-class"
>
  <div
    className="ui-loading-overlay__content"
  >
    Some data
  </div>
</div>
`;

exports[`LoadingOverlay #render() render loading overlay with left message direction 1`] = `
<div
  className="ui-loading-overlay ui-loading-overlay_loading my-custom-class"
>
  <div
    className="ui-loading-overlay__loading-container"
  >
    <[object Object] />
    <span
      classNames="ui-loading-overlay__loading-message ui-loading-overlay__loading-message_left"
    />
  </div>
  <div
    className="ui-loading-overlay__content"
  >
    Data is loading...
  </div>
</div>
`;

exports[`LoadingOverlay #render() render loading overlay with loading message 1`] = `
<div
  className="ui-loading-overlay ui-loading-overlay_loading my-custom-class"
>
  <div
    className="ui-loading-overlay__loading-container"
  >
    <[object Object] />
    <span
      classNames="ui-loading-overlay__loading-message ui-loading-overlay__loading-message_right"
    >
      Loading your data...
    </span>
  </div>
  <div
    className="ui-loading-overlay__content"
  >
    Data is loading...
  </div>
</div>
`;

exports[`LoadingOverlay #render() render loading overlay with transparency 1`] = `
<div
  className="ui-loading-overlay ui-loading-overlay_transparent my-custom-class"
>
  <div
    className="ui-loading-overlay__content"
  >
    Some data
  </div>
</div>
`;

exports[`LoadingOverlay #render() render loading overlay with transparency in loading mode 1`] = `
<div
  className="ui-loading-overlay ui-loading-overlay_loading ui-loading-overlay_transparent my-custom-class"
>
  <div
    className="ui-loading-overlay__loading-container"
  >
    <[object Object] />
    <span
      classNames="ui-loading-overlay__loading-message ui-loading-overlay__loading-message_right"
    />
  </div>
  <div
    className="ui-loading-overlay__content"
  >
    Data is loading... You can see that message.
  </div>
</div>
`;

exports[`LoadingOverlay #render() render without children - should return null 1`] = `""`;

exports[`LoadingOverlay #render() render without spinner 1`] = `
<div
  className="ui-loading-overlay ui-loading-overlay_loading my-custom-class"
>
  <div
    className="ui-loading-overlay__loading-container"
  >
    <span
      classNames="ui-loading-overlay__loading-message ui-loading-overlay__loading-message_right"
    />
  </div>
  <div
    className="ui-loading-overlay__content"
  >
    Data is loading...
  </div>
</div>
`;
